update the description (#2)
DevExpress-Examples/blazor-default-settings: Readme.md

[![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/T827941)
[![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
<!-- default badges end -->

# DevExpress Blazor Components - Set predefined settings for the specific component

> In case of a large web application, controls usually have the same settings on different pages. The recommended Blazor approach is to specify the component settings in the markup. It is possible to override the **SetParametersAsync** method to specify default settings for Blazor components in your application. However, we do not recommend that you use this approach as it is not natural for Blazor applications. We also do not test this approach to specify predefined settings with all our controls, so you can use it with your own risk.

Reviewer: ```suggestion
Controls usually have the same settings for different pages of a large web application, and we recommend that you specify the component settings in markup. You can also override the **SetParametersAsync** method to specify the default settings for Blazor components in your application. However, we do not recommend that you use this approach because we did not test this approach to specify predefined settings with all our controls.

```

Reviewer: - What do you mean when you say `it is not natural`? 
- You do not do something `with your own risk`, you do it **at** your own risk. However, it is unusual to say this in this context since you've already stated that this is an untested approach
- You do not test it, you either **tested** it or **did not** test it since you are making a recommendation here and it refers to a past event

Author: 1. I mean that it is not a common approach for Blazor. Users who used other dotnet platforms may want to use the way described in this example. But it is not the way how they should do this.
2-3. Actually, the whole point of updating this description is to emphasize that a user can but should not use this approach. Allow me to disagree that the recommendation refers to the past event. We never test specifying predefined settings (before, now, and after). So, it means that some day we can change our component so that this approach breaks and a user's app will not work.
The main idea is that the approach is not good, not that our components lack testing.
Can we rewrite as follows?
```suggestion
Controls usually have the same settings for different pages of a large web application, and we recommend that you specify the component settings in markup. You can also override the **SetParametersAsync** method to specify the default settings for Blazor components in your application. However, we do not recommend that you use this approach because it is not a common approach in Blazor. Use it at your own risk.

```

Reviewer: Ok

![Grid with predefined settings and default](images/result.png)

> Create a component that inherits from our Blazor control and overrides the **SetParametersAsync** method. In this method, set the necessary default values for the **public** properties before you call the base method. You can initialize properties with custom default values in the markup because the base method is called after your custom settings. We do not recommend that you override and change values of any protected properties.

Reviewer: - `that inherits` Inherits what?
- `the necessary default values` Define **necessary** 
- in the markup -> in markup

Author: ```suggestion
Create an inheritor from our Blazor control that overrides the **SetParametersAsync** method. In this method, set the desired default values for the **public** properties before you call the base method. You can initialize properties with custom default values in markup because the base method is called after your custom settings. We do not recommend that you override and change values of any protected properties.

```

Reviewer: Create an inheritor from our Blazor control that overrides the **SetParametersAsync** method. In this method, set the default values for the **public** properties before you call the base method. You can initialize properties with custom default values in markup because the base method is called after your custom settings. We do not recommend that you override and change the values of protected properties.

Author: ```suggestion
Create an inheritor from our Blazor control that overrides the **SetParametersAsync** method. In this method, set the default values for the public properties before you call the base method. You can initialize properties with custom default values in markup because the base method is called after your custom settings. We do not recommend that you override and change the values of protected properties.

```

> In this example, the [MyDataGrid](./CS/DxBlazorComponentsDefaultSettings/Components/MyDataGrid.cs) component contains the inherited **DxDataGrid** component whose **PageSize**, **ShowFilterRow**, and **ShowGroupPanel** properties contain new default values. You can compare the **MyDataGrid** component with the default **DxDataGrid** component on the [Index](./CS/DxBlazorComponentsDefaultSettings/Pages/Index.razor) page: they have the same settings in the markup.

Reviewer: ```suggestion
In this example, the [MyDataGrid](./CS/DxBlazorComponentsDefaultSettings/Components/MyDataGrid.cs) component contains the inherited **DxDataGrid** component whose **PageSize**, **ShowFilterRow**, and **ShowGroupPanel** properties contain new default values. You should compare the **MyDataGrid** component with the default **DxDataGrid** component on the [Index](./CS/DxBlazorComponentsDefaultSettings/Pages/Index.razor) page to ensure they have the same settings in markup.

```

Author: I would use something that implies fewer obligations.

```suggestion
In this example, the [MyDataGrid](./CS/DxBlazorComponentsDefaultSettings/Components/MyDataGrid.cs) component contains the inherited **DxDataGrid** component whose **PageSize**, **ShowFilterRow**, and **ShowGroupPanel** properties contain new default values. You can compare the **MyDataGrid** component with the default **DxDataGrid** component on the [Index](./CS/DxBlazorComponentsDefaultSettings/Pages/Index.razor) page to ensure they have the same settings in markup.

```

Reviewer: Ok

> This approach is not applicable to the **DxChartXXXSeries** components and specify the **ValueField**, **ArgumentField**, **NameField**, and **PaneField** properties.

Reviewer: ```suggestion
You cannot use this approach for **DxChartXXXSeries** components and specify the **ValueField**, **ArgumentField**, **NameField**, and **PaneField** properties.

```
